### DISCUSSION TITLE
Are there any plans on porting this over to VS Code?

### DISCUSSION THREAD

**Discussion by aadimator** [marimo-team/marimo#245] [Q&A]
Most of my work is being done in VS Code, even Jupyter Notebooks. I was wondering if there are any plans on porting this over to VS Code, maybe like an Extension or so?

Thanks

**Comment by mscolnick** [comment]
hey @aadimator, we do have some initial work for VS Code integration. you can download our extension here:
https://marketplace.visualstudio.com/items?itemName=marimo-team.vscode-marimo

it's fairly basic and looking for feedback on it before adding more features. as of now:
* you can edit/run marimo notebooks after opening them in VS Code
* there's a marimo panel that shows you all the marimo notebooks in your project directory

we don't use the vscode code editors (or Notebook API) and instead embed the marimo application in a panel. this is so we can still display our rich outputs and keep the same clean/simple UX.

![image](https://github.com/marimo-team/marimo/assets/2753772/06faaed6-939f-43be-b73f-742beff59bd7)

**Comment by julianspaeth** [comment]
Is there also a possibility to run it in PyCharm directly?

**Reply by mscolnick** [reply]
We do not have a PyCharm integration
